legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -386,13 +387,50 @@ else if (matchingQueries.size() == 0)
         query.owner = currentUser;
         query.createdAt = currentQuery.createdAt;
         query.lastUpdatedAt = Instant.now().toEpochMilli();
+        query.originalVersionId = currentQuery.originalVersionId;
         this.getQueryCollection().findOneAndReplace(Filters.eq("id", queryId), queryToDocument(query));
         QueryEvent updatedEvent = createEvent(query.id, QueryEvent.QueryEventType.UPDATED);
         updatedEvent.timestamp = query.lastUpdatedAt;
         this.getQueryEventCollection().insertOne(queryEventToDocument(updatedEvent));
         return query;
     }
 
+    public Query patchQuery(String queryId, Query updatedQuery, String currentUser) throws JsonProcessingException
+    {
+        Query currentQuery = this.getQuery(queryId);
+        // Make sure only the owner can update the query
+        // NOTE: if the query is created by an anonymous user previously, set the current user as the owner
+        if (currentQuery.owner != null && !currentQuery.owner.equals(currentUser))
+        {
+            throw new ApplicationQueryException("Only owner can update the query", Response.Status.FORBIDDEN);
+        }
+
+        Class<? extends Query> queryClazz = currentQuery.getClass();
+        for (java.lang.reflect.Field field : queryClazz.getDeclaredFields())
+        {
+            try
+            {
+                field.setAccessible(true);
+                Object updatedValue = field.get(updatedQuery);
+                if (updatedValue != null)
```

Comment:
this might require some rethinking about if we have the api take from `Query updatedQuery` to something ese
but one use case this current flow does not support is if we want to set  a value to null if it has already been set to something. 
woud using the mongo set operation help here, and update only those fields?

Reply:
If the patchQuery function still takes in a parameter of type Query that we are going to update fields upon, to generate the payload of mongo set operation, I think we'll still need to compose a document by iterating the provided parameter to get non-null fields, then call set? 😅😅

Or,  could you please be more specific about what the parameter type should be? 👀😅😅
Thanks